File: ListViewEditable.cs
using System;
using System.Windows.Forms;

/// <summary>
/// # Will create a listview component that has editiable sub items
///  Ronny Frister gave an example of how to do this, i just added a few lines and added it to a class structure to make it more useable.
/// </summary>
class ListViewEditable : ListView
{
    public TextBox editbox = new TextBox();
    public ListViewHitTestInfo hitInfo;

    public ListViewEditable()
    {
        this.GridLines = true;
        this.FullRowSelect = true;
        this.DoubleBuffered = true;
        editbox.Parent = this;
        editbox.Hide();
        editbox.LostFocus += new EventHandler(Editbox_LostFocus);
        this.MouseDoubleClick += new MouseEventHandler(This_DoubleClick);
    }

    private void Editbox_LostFocus(object sender, EventArgs e)
    {
        hitInfo.SubItem.Text = editbox.Text;
        editbox.Hide();
    }

    private void This_DoubleClick(object sender, MouseEventArgs e)
    {
        hitInfo = this.HitTest(e.X, e.Y);
        editbox.Bounds = hitInfo.SubItem.Bounds;
        editbox.Text = hitInfo.SubItem.Text;
        editbox.Focus();
        editbox.Show();
    }
}

File: ListViewEditable.vb
''' <summary>
''' ListViewEditable will allow a user to edit subitem labels of a listview.
''' </summary>
Public Class ListViewEditable
    Inherits ListView

    Public WithEvents Editbox As New TextBox
    Public HitInfo As ListViewHitTestInfo

    Sub New()
        Me.View = Windows.Forms.View.Details
        Me.GridLines = True
        Me.FullRowSelect = True
        Me.DoubleBuffered = True
        Editbox.Parent = Me
        Editbox.Hide()
    End Sub

    Private Sub ListViewEditable_DoubleClick(sender As Object, e As MouseEventArgs) Handles Me.MouseDoubleClick
        HitInfo = Me.HitTest(e.X, e.Y)
        Editbox.Bounds = HitInfo.SubItem.Bounds
        Editbox.Text = HitInfo.SubItem.Text
        Editbox.Focus()
        Editbox.Show()
    End Sub

    Private Sub Editbox_LostFocus(sender As Object, e As EventArgs) Handles Editbox.LostFocus
        HitInfo.SubItem.Text = Editbox.Text
        Editbox.Hide()
    End Sub
End Class
